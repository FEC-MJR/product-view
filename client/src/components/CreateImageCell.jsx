import React from 'react'
import styled from 'styled-components'


const StyledThumb = styled.img`
  width: 60px;
  height: 60px;
  background: #fff;
  margin: 5px;
  border: ${props => {
    if (props.props.index === props.props.activeCell) {
      return '1px solid black'
    }
  }}
`;

//creates image cell from props passed down from <SliderThumb/>
const CreateImageCell = (props) => {

  return(
    <StyledThumb className={`thumbnail-${props.index}`} props={props} src={props.imageSource}></StyledThumb>
  )
}

export default CreateImageCell