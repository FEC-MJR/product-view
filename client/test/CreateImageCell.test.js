import React from 'react'
import { shallow, mount, render } from 'enzyme'
import renderer from 'react-test-renderer'
import CreateImageCell from '../src/components/CreateImageCell.jsx'

describe("Component testing for <CreateImageCell/>", () => {

  const props = {
    data:  {
      currentProduct: {
        name: "KANKEN LAPTOP 13\"",
        type: "backpack",
        description: "Sed eu eros. Nam consequat dolor vitae dolor. Donec fringilla. Donec feugiat metus sit",
        userRating: 5,
        totalRatings: 6,
        price: 110.00,
        gender: "Unisex",
        style: "Style # F27171",
        size: ["One Size"],
        color: "166 - Acorn",
        image: ["https://hrphxfec.s3-us-west-1.amazonaws.com/Kanken+Laptop+13/kanken_acorn.jpg"]
      }
    }
  }

  test("<CreateImageCell/> should render", () => {

    const wrapper = shallow(<CreateImageCell {...props}/>)
    expect(wrapper)
  })

  test("<CreateImageCell/> snapshot", () => {

    const tree = renderer.create(<CreateImageCell {...props}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})