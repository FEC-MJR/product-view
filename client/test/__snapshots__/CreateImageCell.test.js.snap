// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component testing for <CreateImageCell/> <CreateImageCell/> snapshot 1`] = `
<img
  className="sc-AxjAm jLqTMr"
/>
`;
